apps/site/assets/ts/app/channels.ts

```diff
@@ -84,22 +84,41 @@ const leaveChannel = (id: string): void => {
 
 const setupChannels = (): void => {
   window.socket = new Socket("/socket", {});
+  window.socket.onClose(event => {
```

Comment:
question: Is the `close` event type only ever caused by the socket being closed by the browser?

Reply:
probably not! I would guess it _could_ also be closed programmatically (we don't do it as far as I'm aware). and honestly, I'm not sure where/how else this event could be caused.